pkg/controller/master/schedulevmbackup/common.go

```diff
@@ -0,0 +1,465 @@
+package schedulevmbackup
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"time"
+
+	"go.uber.org/multierr"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/controller/master/backup"
+	"github.com/harvester/harvester/pkg/ref"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	scheduleVMBackupKindName = "ScheduleVMBackup"
+	timeFormat               = "20060102.1504"
+
+	reachMaxFailure = "Reach Max Failure"
+
+	proactiveSuspend = "Proactive Scheulde Suspend"
+)
+
+const (
+	updateInterval    = 5 * time.Second
+	reconcileInterval = 5 * time.Second
+
+	svmbackupPrefix = "svmb"
+
+	cronJobNamespace    = "harvester-system"
+	cronJobBackoffLimit = 3
+	cronJobCmd          = "sleep"
+	cronJobArg          = "10"
+)
+
+var scheduleVMBackupKind = harvesterv1.SchemeGroupVersion.WithKind(scheduleVMBackupKindName)
+
+func cronJobName(svmbackup *harvesterv1.ScheduleVMBackup) string {
+	return fmt.Sprintf("%s-%s", svmbackupPrefix, svmbackup.UID)
+}
+
+func getCronJob(h *svmbackupHandler, svmbackup *harvesterv1.ScheduleVMBackup) (*batchv1.CronJob, error) {
+	cronJob, err := h.cronJobCache.Get(cronJobNamespace, cronJobName(svmbackup))
+	if err != nil {
+		return nil, err
+	}
+
+	return cronJob, nil
+}
+
+func vmBackupName(svmbackup *harvesterv1.ScheduleVMBackup, timestamp string) string {
+	return fmt.Sprintf("%s-%s-%s", svmbackupPrefix, svmbackup.UID, timestamp)
+}
+
+func getVMBackup(h *svmbackupHandler, svmbackup *harvesterv1.ScheduleVMBackup, timestamp string) (*harvesterv1.VirtualMachineBackup, error) {
+	vmBackup, err := h.vmBackupCache.Get(svmbackup.Namespace, vmBackupName(svmbackup, timestamp))
+	if err != nil {
+		return nil, err
+	}
+
+	return vmBackup, nil
+}
+
+func currentVMBackups(h *svmbackupHandler, svmbackup *harvesterv1.ScheduleVMBackup) (
+	[]*harvesterv1.VirtualMachineBackup, []*harvesterv1.VirtualMachineBackup,
+	*harvesterv1.VirtualMachineBackup, int, error) {
```

Comment:
Because there's several return parameters here with the same or similar types, I'd suggest naming them, to make it clear what they're for when reading the function definition, something like:

```suggestion
func currentVMBackups(h *svmbackupHandler, svmbackup *harvesterv1.ScheduleVMBackup) (
	vmbackups []*harvesterv1.VirtualMachineBackup, errVMBackups []*harvesterv1.VirtualMachineBackup,
	lastVMBackup *harvesterv1.VirtualMachineBackup, failure int, err error) {
```

(If you take this change, it will mean also replacing a bunch of `:=` with just `=` in the function body)

Reply:
Updated per suggestion, thanks.